connectors/services/job_scheduling.py

```diff
@@ -167,29 +170,40 @@ async def _run(self):
                 await self.sync_job_index.close()
         return 0
 
-    async def _scheduled_sync(self, connector, job_type):
+    async def _try_schedule_sync(self, connector, job_type):
+        last_wake_up_time = self.last_wake_up_time
+        this_wake_up_time = datetime.utcnow()
+
+        print(
+            f"Last time woke up at {last_wake_up_time}, woke up at {this_wake_up_time}"
+        )
+
         @with_concurrency_control()
-        async def _should_schedule_scheduled_sync(job_type):
+        async def _should_schedule(job_type):
             try:
                 await connector.reload()
             except DocumentNotFoundError:
                 connector.log_error("Couldn't reload connector")
                 return False
 
             job_type_value = job_type.value
-            now = datetime.utcnow()
+
             last_sync_scheduled_at = connector.last_sync_scheduled_at_by_job_type(
                 job_type
             )
 
-            if last_sync_scheduled_at is not None and last_sync_scheduled_at > now:
+            if (
+                last_sync_scheduled_at is not None
+                and last_sync_scheduled_at > this_wake_up_time
+            ):
                 connector.log_debug(
                     f"A scheduled '{job_type_value}' sync is created by another connector instance, skipping..."
                 )
                 return False
 
             try:
-                next_sync = connector.next_sync(job_type)
+                next_sync = connector.next_sync(job_type, last_wake_up_time)
+                print(f"Next sync is at {next_sync}")
```

Comment:
```suggestion
```

Probably also a debug statement? Or should we change this to use `logger`?

Reply:
Yup, good call